src/azure-cli/azure/cli/command_modules/network/_template_builder.py

```diff
@@ -121,7 +121,15 @@ def _ag_subresource_id(_type, name):
 
     frontend_ip_configs = []
 
-    if private_ip_address is not None or public_ip_id is None:
+    if public_ip_id:
+        frontend_public_ip = _build_frontend_ip_config(cmd, frontend_public_ip_name,
+                                                       public_ip_id=public_ip_id,
+                                                       enable_private_link=enable_private_link,
+                                                       private_link_configuration_id=private_link_configuration_id)
+        frontend_ip_configs.append(frontend_public_ip)
+
+        frontend_ip_config_id = _ag_subresource_id('frontendIPConfigurations', frontend_public_ip_name)
+    if private_ip_address != '' or public_ip_id is None:
         enable_private_link = False if public_ip_id else enable_private_link
```

Comment:
>enable_private_link [](start = 57, length = 19)

Change the default value for private_ip_address to None?

Reply:
Changed as you said

Reply:
Add comments to describe the logic here as it's a little bit tricky.

---
In reply to: [518600363](https://github.com/Azure/azure-cli/pull/15757#discussion_r518600363) [](ancestors = 518600363)